### PR TITLE
join migrated project

### PR DESCRIPTION
**Pull request by weswalla**
joining an already migrated project works, however, it doesn't show up in the list of syncing projects. It either shows nothing, or the project once it is loaded @Connoropolous

### CODE REVIEW

File: electron/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "acorn",
-  "version": "4.0.0-alpha",
+  "version": "3.0.0-alpha",
```
Comment (Connoropolous) [inline electron/package.json:3]: revert

File: electron/src/holochain.ts [hunk 2]
```diff
@@ -65,14 +65,14 @@ const COMMUNITY_PROXY_URL =
 const devOptions: ElectronHolochainOptions = {
   happPath: profilesHappPath, // preload
   datastorePath: process.env.ACORN_TEST_USER_2
-    ? '../user2-data/databases'
-    : path.join(__dirname, '../../user-data/databases'),
+    ? path.join(__dirname, `../../user2-data/databases-${INTEGRITY_VERSION_NUMBER}`)
+    : path.join(__dirname, `../../user-data/databases-${INTEGRITY_VERSION_NUMBER}`),
```
Comment (Connoropolous) [inline electron/src/holochain.ts:69]: can you push a step further on this refactor and consolidate this with the other references like we talked about? add another file called `paths` or some such like name and export constants from it that this file and the other import from.

File: electron/src/index.ts [hunk 3]
```diff
@@ -217,14 +217,21 @@ ipcMain.on('initiateUpdate', () => {
...
+const dataPath = app.isPackaged ? app.getPath('userData') : path.join(__dirname, `../../user${userNumber}-data/`)
+const migrationFile = path.join(
+  dataPath,
+  `${MIGRATION_FILE_NAME_PREFIX}${INTEGRITY_VERSION_NUMBER}`
+)
+const prevIntegrityVersion = INTEGRITY_VERSION_NUMBER - 1
+const prevVersionMigrationFile = path.join(
+  dataPath,
+  `${MIGRATION_FILE_NAME_PREFIX}${prevIntegrityVersion}`
+)
```
Comment (Connoropolous) [inline electron/src/index.ts:232]: for best practice by convention, we could make all these global consts named using CAPITAL_UNDERSCORE casing. When reading functions, it helps to be able to quickly distinguish between local and global variables.

File: electron/src/holochain.ts [hunk 4]
```diff
@@ -65,14 +65,14 @@ const COMMUNITY_PROXY_URL =
...
   appWsPort: process.env.ACORN_TEST_USER_2 ? 8899 : 8888,
   adminWsPort: process.env.ACORN_TEST_USER_2 ? 1236 : 1234,
   keystorePath: process.env.ACORN_TEST_USER_2
-    ? '../user2-data/keystore'
-    : path.join(__dirname, '../../user-data/keystore'),
+    ? path.join(__dirname, `../../user2-data/keystore-${KEYSTORE_VERSION_NUMBER}`)
+    : path.join(__dirname, `../../user-data/keystore-${KEYSTORE_VERSION_NUMBER}`),
   passphrase: 'test-passphrase',
   proxyUrl: COMMUNITY_PROXY_URL,
 }
```
Comment (Connoropolous) [inline electron/src/holochain.ts:78]: at the very end of this file, there is this:
`export { projectsHappPath as projectsDnaPath,` 
can you get rid of the projectsDnaPath alias?

**Review by Connoropolous — CHANGES_REQUESTED** [review]
per your comment in the PR description, we may need to dispatch a redux action after the project has been joined. Take a look at the other examples of project creation, joining, or importing to see what that action to dispatch might be.

File: electron/src/paths.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { app } from 'electron'
+import * as path from 'path'
+import { INTEGRITY_VERSION_NUMBER } from './holochain'
```
Comment (weswalla) [inline electron/src/paths.ts:3]: @Connoropolous when I test this out `INTEGRITY_VERSION_NUMBER` is `undefined` at runtime.

Comment (Connoropolous) [reply electron/src/paths.ts:3]: Maybe it’s a circular dependency. We could move those consts into this file

**Review by Connoropolous — APPROVED** [review]
Sweet

**Pull request merged**